frontend/appflowy_flutter/lib/plugins/database_view/calendar/presentation/toolbar/calendar_toolbar.dart

```diff
@@ -81,3 +80,86 @@ class _SettingButtonState extends State<_SettingButton> {
   }
 }
+
+class _UnscheduleEventsButton extends StatefulWidget {
+  const _UnscheduleEventsButton({Key? key}) : super(key: key);
+
+  @override
+  State<_UnscheduleEventsButton> createState() =>
+      _UnscheduleEventsButtonState();
+}
+
+class _UnscheduleEventsButtonState extends State<_UnscheduleEventsButton> {
+  late PopoverController _popover;
+
+  @override
+  void initState() {
+    _popover = PopoverController();
+    super.initState();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return BlocBuilder<CalendarBloc, CalendarState>(
+      builder: (context, state) {
+        final unscheduledEvents = state.allEvents
+            .where((e) => e.date == DateTime.fromMillisecondsSinceEpoch(0));
+        final viewId = context.read<CalendarBloc>().viewId;
+        final rowCache = context.read<CalendarBloc>().rowCache;
+        return AppFlowyPopover(
+          direction: PopoverDirection.bottomWithCenterAligned,
+          controller: _popover,
+          offset: const Offset(0, 8),
+          child: FlowyTextButton(
+            "${LocaleKeys.calendar_settings_noDateTitle.tr()} (${unscheduledEvents.length})",
+            fillColor: Colors.transparent,
+            hoverColor: AFThemeExtension.of(context).lightGreyHover,
+            padding: GridSize.typeOptionContentInsets,
+          ),
+          popupBuilder: (context) {
+            final cells = unscheduledEvents
+                .map(
+                  (CalendarEventData<CalendarDayEvent> event) => SizedBox(
+                    height: GridSize.popoverItemHeight,
+                    child: FlowyTextButton(
+                      event.title,
+                      fillColor: Colors.transparent,
+                      hoverColor: AFThemeExtension.of(context).lightGreyHover,
+                      padding: GridSize.typeOptionContentInsets,
+                      onPressed: () {
+                        showEventDetails(
+                          context: context,
+                          event: event.event!,
+                          viewId: viewId,
+                          rowCache: rowCache,
+                        );
+                        _popover.close();
+                      },
+                    ),
+                  ),
+                )
+                .toList();
+            if (cells.isEmpty) {
+              return SizedBox(
+                height: GridSize.popoverItemHeight,
+                child: Center(
+                  child: FlowyText.medium(
+                    LocaleKeys.calendar_settings_emptyNoDate.tr(),
+                    color: Theme.of(context).hintColor,
+                  ),
+                ),
+              );
+            }
+            return ListView.separated(
+              itemBuilder: (context, index) => cells[index],
+              itemCount: cells.length,
+              separatorBuilder: (context, index) =>
+                  VSpace(GridSize.typeOptionSeparatorHeight),
+              shrinkWrap: true,
```

Comment:
shrinkwrap is expensive as it renders the entire ListView, not just the widgets available near the viewport. Could you find a way to not use it?

Reply:
I don't know how we could do that. The ListView could contain anywhere between 1 item or 100 items and the popover's height depends on the number of items. The same occurs for CustomScrollView and SliverLists. Do you have a suggestion on how to achieve this?

Reply:
If the popover's size depends on the number of widgets inside, does the popover extend past the window's viewport if there are, say, 100 items? 

Regardless, I think adding bounds to the popover size could be reasonable rather than shrink-wrapping. 

cc @Xazin

Reply:
No, the popover won't extend past the edges of the window. Since all that space until the edges of the window will all be consumed by the ListView without shrinkWrap, adding a maxHeight definitely seems reasonable. However, if there's only one item and we'll still end up with extra whitespace.

Reply:
Also the popoverBuilder is put into a Container with box constraints. So it will never be bigger than 200x600 (I can't remember exact numbers - see AppFlowyPopover).

As long as the text has overflow so if it is longer than the 200 width it will show ellipsis it should be fine.

Reply:
Awesome, I'm resolving this thread